src/multiple_shooting_objective.jl

```diff
@@ -0,0 +1,50 @@
+export MultipleShootingObjective, multiple_shooting_objective
+
+immutable MultipleShootingObjective{F,C} <: Function
+  multiple_shooting_cost_function::F
+  constraints::C
+end
+
+my_problem_new_parameters = function (prob::ODEProblem,p,t1,t2,data)
+  f = (t,u,du) -> prob.f(t,u,p,du)
+  uEltype = eltype(p)
+  tspan = (uEltype(t1),uEltype(t2))
+  u0 = [uEltype(data[:,t1][i]) for i in 1:length(prob.u0)]
+  ODEProblem(prob.f,u0,(t1,t2))
+end
+
+
+function multiple_shooting_objective(prob::DEProblem,alg,loss,timestamp=nothing;mpg_autodiff = false,
+                              verbose_opt = false,verbose_steps = 100,
+                              prob_generator = problem_new_parameters,
+                              kwargs...)
+  if verbose_opt
+    count = 0 # keep track of # function evaluations
+  end
+  timestamps = sort(sample(2:length(t)-1, 10, replace = false))
```

Comment:
random isn't necessarily good here. The default should probably just be a `linspace`.